### DISCUSSION TITLE
Missing features in osu! chat

### DISCUSSION THREAD

**Discussion by vatei** [ppy/osu#29129] [Ideas]
As mentioned on the osu! discord before, the osu! chat is missing some features, making it feel barebones and less legible.
Here are some of them I think are important:

- Highlight mentions (color or bold font)
   - Additional highlight when receiving a mention (maybe using a fading highlight animation)
   
   ![osu-chat](https://github.com/user-attachments/assets/567f86b3-26a6-49f8-8466-52a5d0f989bf)

   The existing notification could be used only when the chat isn't visible

-  Autocomplete mentions using tab
- Block user
- Mute user (prevents notifications)
- Right click menu on messages
    - Mention (same as the existing one in the username right click menu)
    - Reply (would quote the message or part of it if it’s too long. Doesn't need to be added to the chat box, maybe with a "reply"/"replying to user" indicator and the mention, then the quote is displayed after sending a message)
    
     ![image](https://github.com/user-attachments/assets/f93849f8-31d2-4199-a564-9140ef13ced1)
     ![image](https://github.com/user-attachments/assets/99ccc7e4-b902-4ab8-bcc8-eae84b84ba59)

    - Report message (meaning the content of the message would be appended to the report)
- Ability to change the color of the username (or matching it to the profile hue) 
Partially discussed in https://github.com/ppy/osu/discussions/23145#discussioncomment-10162170

- Accessing the daily challenge chat via #daily-challenge (useful for chatting without having the daily challenge screen open)

I included some paint mockups to explain some points, hope that makes it clearer
It's mainly QoL, so I hope some of them can be implemented, thanks again for reading!

Edit: added a few more ideas in the comments

**Comment by bdach** [comment]
also mentioned in chat:

- alternating chat line colours for accessibility
- (tba more if i get told more)

**Comment by peppy** [comment]
Mention highlighting and alternating line colours have been implemented.

**Comment by vatei** [comment]
- Being able to do !roll in chat is also missing (it's dumb but I miss that)
- Ordering direct messages by last message ( https://github.com/ppy/osu/discussions/18461 )
- Being able to collapse the "channels" or "direct messages" sections in the chat window

**Reply by vatei** [reply]
Bumping for collapsing sections as it's even messier now with team chat added

**Comment by pigeonskew** [comment]
Having a #daily-challenge chat accessible outside of daily challenge can spoil the map to others that hasn't entered the map of the day yet. Even if we block the /np command in that chat, players could just type the name of the map themselves. 

Also, that allows players that possibly don't even play daily challenge to chat, making it some any other channel that already exists in the game. Might be trivial to others but I think having an element of surprise and exclusivity of its chat makes daily challenge fun, at least for me.

**Reply by vatei** [reply]
You're not wrong, I didn't consider that first point
Although to access this chat you would have to purposely add it to your channels if it's not default, I feel like you would know what you're getting into
